### PR TITLE
[PM-1655] Trial Registration Layout

### PR DESCRIPTION
**Pull request by nick-livefront**
Server related changes: https://github.com/bitwarden/server/pull/4221

## Objective

Migrate from the `TrialInitiationComponent` to a new component that utilizes the `AnonLayoutWrapperComponent`. All existing marketing content will be handled via CMS.

### Open Questions

- The `SubscriptionProduct` enum & the new `Product` enum I added here represent the same thing in different ways. Another refactor opportunity possibly? 

## Code changes

- **CompleteTrialInitiationComponent**
  - New sign up component, heavily based off of `TrialInitiationComponent`
  - Uses `RegistrationFinishService` to validate the user registration
  - For non-trial users, links the user back directly to the login screen
  - Consolidated secrets manager logic here were applicable. 
- Added `maxWidth` to the AnonLayoutComponent in order to match the designs for the trial component. 
- **free-trial-text.resolver.ts:**
  - Angular [Resolver](https://angular.io/api/router/ResolveFn) to resolve the `pageTitle` passed to the `AnonLayoutWrapperComponent`
- **confirmation-details.component.html:**
  - Allow for both SecretsManager & PasswordManager to use the same confirmation message

### Password Manager
|Families|Teams|Enterprise|
|-|-|-|
|<video src="https://github.com/user-attachments/assets/b10c018c-2fc2-4895-92fb-61cd975302db" />|<video src="https://github.com/user-attachments/assets/b91030c6-99b2-44aa-badc-bb20a223ae7a"/>|<video src="https://github.com/user-attachments/assets/93d06da3-3e78-49fb-b57b-5421205b9f09"/>|

### Secrets Manager
|Teams|Enterprise|
|-|-|
|<video src="https://github.com/user-attachments/assets/3513a1d6-8642-4bab-a1f4-ece7370f3c32" />|<video src="https://github.com/user-attachments/assets/739270d6-b48c-42d2-9911-3b25e5d81e31" />|

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

**Comment by micahblut** [comment]
@nick-livefront one question above about an error message.
Also, the "Create account" button doesn't stretch the width of the form in the designs.

**Comment by nick-livefront** [comment]
> @nick-livefront one question above about an error message. Also, the "Create account" button doesn't stretch the width of the form in the designs.

@micahblut This is configured by the underlying component. Danielle gave the go-ahead to continue as is and this can be fixed in the component. [Figma discussion](https://www.figma.com/design/81uGWWdQVAQfKDiR4qltoq?node-id=3105-6540#875174631).

File: libs/auth/src/angular/input-password/password-input-result.ts [hunk 1]
```diff
@@ -6,4 +6,5 @@ export interface PasswordInputResult {
   masterKeyHash: string;
   kdfConfig: PBKDF2KdfConfig;
   hint: string;
+  password: string;
```
Comment (nick-livefront) [inline libs/auth/src/angular/input-password/password-input-result.ts:9]: In order to automatically login a user, I needed master password. I followed [the pattern](https://github.com/bitwarden/clients/blob/36487d69ba64d2d4757c73fd64e48ac4993d7318/apps/web/src/app/auth/trial-initiation/complete-trial-initiation/complete-trial-initiation.component.ts#L297-L306) that the current Trial Initiation component uses.

Is there a better approach?

File: libs/auth/src/angular/anon-layout/anon-layout.component.ts [hunk 2]
```diff
@@ -23,6 +23,12 @@ export class AnonLayoutComponent {
   @Input() subtitle: string;
   @Input() icon: Icon;
   @Input() showReadonlyHostname: boolean;
+  /**
+   * Max width of the layout content
+   *
+   * @default 'md'
+   */
+  @Input() maxWidth: "md" | "3xl" = "md";
```
Comment (nick-livefront) [inline libs/auth/src/angular/anon-layout/anon-layout.component.ts:31]: There are some differing designs for the anon layout and max-widths.

@rr-bw you added the original max width to the layout here: https://github.com/bitwarden/clients/pull/9272, any concerns with moving to an input?

File: apps/web/src/app/oss-routing.module.ts [hunk 3]
```diff
@@ -372,6 +374,28 @@ const routes: Routes = [
...
+        component: CompleteTrialInitiationComponent,
+        resolve: {
+          pageTitle: freeTrialTextResolver,
+        },
+        data: {
+          maxWidth: "3xl",
+        } satisfies AnonLayoutWrapperData,
+      },
+      {
+        path: "secrets-manager-trial-initiation",
```
Comment (nick-livefront) [inline apps/web/src/app/oss-routing.module.ts:389]: @cturnbull-bitwarden You have multiple routing paths in [your server PR](https://github.com/bitwarden/server/blob/11e8c8a9012fa4eb745a59931cc2a9db21e494e1/src/Core/Billing/Models/Mail/TrialInititaionVerifyEmail.cs#L29-L33), if I'm interpreting that correctly. 

Here I have them routing to the same component. Do we need multiple paths when the query params will control which product is being trialed?

File: libs/auth/src/angular/anon-layout/anon-layout.component.ts (hunk 2 above)
Comment (JaredSnider-Bitwarden) [reply libs/auth/src/angular/anon-layout/anon-layout.component.ts:31]: Per discussion with @danielleflinn from design, we are good with proceeding with this input approach. Thank you!

File: libs/auth/src/angular/input-password/password-input-result.ts (hunk 1 above)
Comment (JaredSnider-Bitwarden) [reply libs/auth/src/angular/input-password/password-input-result.ts:9]: No issues with this approach!

**Review by JaredSnider-Bitwarden — DISMISSED** [review]
Auth changes look good! Nice work!

**Review by cturnbull-bitwarden — DISMISSED** [review]
Looks good! Thank you!

File: libs/auth/src/angular/anon-layout/anon-layout-wrapper.component.ts [hunk 4]
```diff
@@ -65,6 +67,7 @@ export class AnonLayoutWrapperComponent implements OnInit, OnDestroy {
 
     this.pageIcon = firstChildRouteData["pageIcon"];
     this.showReadonlyHostname = firstChildRouteData["showReadonlyHostname"];
+    this.maxWidth = firstChildRouteData["maxWidth"];
```
Comment (cturnbull-bitwarden) [inline libs/auth/src/angular/anon-layout/anon-layout-wrapper.component.ts:70]: @nick-livefront  It looks like a merge conflict drifted in from `main`. Would you mind resolving?

Comment (nick-livefront) [reply libs/auth/src/angular/anon-layout/anon-layout-wrapper.component.ts:80]: @cturnbull-bitwarden  [081ef0b](https://github.com/bitwarden/clients/pull/9091/commits/081ef0b8bcf9f662a73b8db667af5eb01f19a2cd) Looks like my merge went sideways.. a bunch of formatting changes were applied on commit. 

Let me see if I can fix it

**Review by cturnbull-bitwarden — APPROVED** [review]

File: libs/auth/src/angular/anon-layout/anon-layout-wrapper.component.ts (hunk 4 above)
Comment (vleague2) [reply libs/auth/src/angular/anon-layout/anon-layout-wrapper.component.ts:80]: There was an update to prettier in main, so you will need to `npm ci` and then `npm run prettier` on your branch to get the formatting back to matching main. (Source -- I did the same thing to my branch on Friday lol)

Comment (cturnbull-bitwarden) [reply libs/auth/src/angular/anon-layout/anon-layout-wrapper.component.ts:80]: Whoops, I had forgotten we recently updated `prettier` https://github.com/bitwarden/clients/pull/10228. Try running `npm run prettier` from the base clients directory

**Comment by nick-livefront** [comment]
My apologies everyone, turns out I had an old version of prettier and needed to update. That caused a bunch of files to be changed.

[35bb619](https://github.com/bitwarden/clients/pull/9091/commits/35bb6194eba30af4386acc167eb6d7cba6b1092b)

File: libs/auth/src/angular/anon-layout/anon-layout-wrapper.component.ts (hunk 4 above)
Comment (nick-livefront) [reply libs/auth/src/angular/anon-layout/anon-layout-wrapper.component.ts:80]: 😓 Just saw the same, thank you!

**Review by cturnbull-bitwarden — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Pull request merged**